applications/SwimmingDEMApplication/custom_utilities/L2_error_projection_utility.h

```diff
@@ -47,10 +47,23 @@ L2ErrorProjection()
 
 virtual ~L2ErrorProjection(){}
 
+void ComputeDofsErrors(ModelPart& r_model_part)
+{
+    const unsigned int n_nodes = r_model_part.Nodes().size();
+    for (unsigned int inode = 1; inode <= n_nodes; ++inode){
+
+        auto& r_vectorial_error = r_model_part.GetNode(inode).FastGetSolutionStepValue(VECTORIAL_ERROR);
+        r_vectorial_error = r_model_part.GetNode(inode).FastGetSolutionStepValue(VELOCITY) - r_model_part.GetNode(inode).FastGetSolutionStepValue(EXACT_VELOCITY);
+
+        auto& r_scalar_error = r_model_part.GetNode(inode).FastGetSolutionStepValue(SCALAR_ERROR);
+        r_scalar_error = r_model_part.GetNode(inode).FastGetSolutionStepValue(PRESSURE) - r_model_part.GetNode(inode).FastGetSolutionStepValue(EXACT_PRESSURE);
+    }
+}
```

Comment:
ouch this is completely wrong
- using `GetNode` is access by Id not by index
- using `GetNode` is a `find` each time, aka you search the same node 6 times per iteration!!!

I recommend using the parallel utilities (in pseudo code, please search the code for "block_for_each" to see usage examples):
~~~c++
block_for_each(r_model_part.Nodes(), [](Node<3>& rNode){
// here put your code
})
~~~

Reply:
Thanks @philbucher, I've already used this parallel utility in other scripts, I will apply this technique to the L2 projection as well.